.github/workflows/test.yml

```diff
@@ -0,0 +1,78 @@
+name: Test
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+jobs:
+  test-unit:
+    name: Unit Tests
+    strategy:
+      matrix:
+        go-version: [ 1.16.x ]
+        platform: [ ubuntu-latest ]
+    runs-on: ${{ matrix.platform }}
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - name: Install Go ${{ matrix.go-version }}
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+        id: go
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: go mod download
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
```

Comment:
cache key on `go.sum`? I wonder if it makes sense to do it on go.mod. If I understand correctly, go.mod is enough for reproducible builds. `go.sum` only adds hash of golang packages to integrity purposes.

Reply:
Might make sense on that level too, I've extracted the caching example from https://github.com/actions/cache/blob/main/examples.md#linux

Reply:
these guys know more than me, so good to go.